src/themis/secure_session.c

```diff
@@ -101,6 +102,17 @@ themis_status_t secure_session_init(secure_session_t* session_ctx,
     soter_status_t soter_status;
     themis_status_t res = THEMIS_SUCCESS;
 
+    if (!sign_key || !sign_key_length) {
+        res = THEMIS_INVALID_PARAMETER;
+        goto err;
+    }
```

Comment:
Doesn't `themis_get_asym_key_kind()` already check for this?

Reply:
Yes, it does. My bad. In this case, should I leave [all the checks in secute_session_peer_init](https://github.com/iamnotacake/themis/blob/anatolii-T1739-runtime-check-ec-keys-in-secure-session/src/themis/secure_session_peer.c#L45-L47) as they are?

Reply:
Well, I wouldn't really mind if key checks are replaced with `themis_get_asym_kind()` call, given that it already performs the checks. However, I wouldn't mind as well if they are left as is, just to minimize the diff. *Actually*, I don't think they are that much of a performance loss here. I (personally) care more about readability than petty optimizations like this. [There was a discussion about that once](https://github.com/cossacklabs/themis/issues/547).